src/store.js

```diff
@@ -18,35 +20,35 @@ export class Store {
         const stringifiedContacts = localStorage.getItem('persistedContacts');
         const persistedContacts = JSON.parse(stringifiedContacts);
 
-        if (!stringifiedState) {
-            if (persistedContacts) {
-                return configureStore({
-                    contacts: Object.assign({}, persistedContacts),
-                });
-            }
+        const initialState = {};
+  
+        if (stringifiedState) {
+            const persistedState = JSON.parse(stringifiedState);
 
-            return configureStore();
-        }
-
-        let persistedState = JSON.parse(stringifiedState);
-
-        persistedState = Object.assign({},
-            persistedState,
-            {
-                transactions: Object.assign({}, persistedState.transactions, {
+            // ignore outdated cache
+            if (persistedState.cacheVersion === CACHE_VERSION) {
+                initialState.transactions = Object.assign({}, persistedState.transactions, {
                     entries: new Map(persistedState.transactions.entries)
-                }),
-                wallets: Object.assign({}, initialWalletState, persistedState.wallets, {
+                });
+                initialState.wallets = Object.assign({}, initialWalletState, persistedState.wallets, {
                     wallets: new Map(persistedState.wallets ? persistedState.wallets.wallets : []),
                     accounts: new Map(persistedState.wallets ? persistedState.wallets.accounts: [])
-                }),
-                network: Object.assign({}, initialNetworkState, persistedState.network),
-                settings: Object.assign({}, initialSettingsState, persistedState.settings),
-                contacts: Object.assign({}, persistedState.contacts || {}, persistedContacts || {}),
+                });
+                initialState.network = Object.assign({}, initialNetworkState, persistedState.network);
+                initialState.settings = Object.assign({}, initialSettingsState, persistedState.settings);
+                
+                // support legacy version of persisted contacts
+                if (persistedState.contacts) {
+                    initialState.contacts = Object.assign({}, persistedState.contacts);
+                }
             }
-        );
+        }
+
+        if (persistedContacts) {
+            initialState.contacts = Object.assign({}, persistedContacts);
```

Comment:
There might be a reason why the code before was handling the contacts as inheriting from each other
```
Object.assign({}, persistedState.contacts || {}, persistedContacts || {})
```
I can't think of a reason now, but there was propably an edge case, maybe?

Would it be possible to do this inheritance here, too?

Reply:
Don't know why you did it like that, but I'm pretty sure it's not necessary. Telling from https://github.com/nimiq/safe/commit/de849a882c0208fa3c7523b5771a3d5ce8f84787 there was never the possibility of both variants coexisting.